### PR TITLE
Refactor: Move some classes from `sql` to `processing` & `server` for reusability

### PR DESCRIPTION
**Pull request by abhishekrb19**
This PR contains non-functional / refactoring changes of the following classes in the `sql` module:

1. Move `ExplainPlan` and `ExplainAttributes` from`sql/src/main/java/org/apache/druid/sql/http`  to `processing/src/main/java/org/apache/druid/query/explain`
2. Move `sql/src/main/java/org/apache/druid/sql/SqlTaskStatus.java` -> `processing/src/main/java/org/apache/druid/query/http/SqlTaskStatus.java`
3. Add a new class `processing/src/main/java/org/apache/druid/query/http/ClientSqlQuery.java` that is effectively a thin POJO version of `SqlQuery` in the `sql` module but without any of the Calcite functionality and business logic.
4. Move `BrokerClient`, `BrokerClientImpl` and `Broker` classes from `sql/src/main/java/org/apache/druid/sql/client` to `server/src/main/java/org/apache/druid/client/broker`.
5. Remove `BrokerServiceModule` that provided the `BrokerClient`. The functionality is now contained in `ServiceClientModule` in the `server` package itself which provides all the clients as well.

This is done so that we can reuse the said classes in https://github.com/apache/druid/pull/17353 without brining in Calcite  and other dependencies to the Overlord.

This PR has:

- [x] been self-reviewed.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/src/main/java/org/apache/druid/sql/http/SqlQuery.java [hunk 1]
```diff
@@ -28,13 +28,17 @@
...
 import javax.annotation.Nullable;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
+/**
+ * @see ClientSqlQuery for the equivalent POJO class for use on the client side.
+ */
 public class SqlQuery
```
Comment (kfaraz) [inline sql/src/main/java/org/apache/druid/sql/http/SqlQuery.java:42]: Can we have this class extend `ClientSqlQuery`?
Alternatively, maybe we could move this class itself to processing module. Is any field in this class stopping us from doing that?

File: processing/src/test/java/org/apache/druid/query/http/ClientSqlQueryTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.druid.jackson.DefaultObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ClientSqlQueryTest
+{
+  @Test
+  public void testSerde() throws JsonProcessingException
```
Comment (kfaraz) [inline processing/src/test/java/org/apache/druid/query/http/ClientSqlQueryTest.java:31]: There should be a test in `SqlQueryTest` to test inter-conversion between `SqlQuery` and `ClientSqlQuery`.

**Review by kfaraz — COMMENTED** [review]
Looks good, minor query regarding `SqlQuery` class.

File: processing/src/test/java/org/apache/druid/query/http/ClientSqlQueryTest.java (hunk 2 above)
Comment (abhishekrb19) [reply processing/src/test/java/org/apache/druid/query/http/ClientSqlQueryTest.java:32]: Done. Also, fixed a related bug with parameters caught in the unit test.

**Review by kfaraz — APPROVED** [review]
LGTM

**Pull request merged**